### PR TITLE
Change custom Event Grid eventType property from an int to a string

### PR DESCRIPTION
**Pull request by TsuyoshiUshio**
This pull request is for the https://github.com/Azure/azure-functions-durable-extension/issues/285 . 

I change EventGridPayload eventType -> runtimeStatus. 

Here is an example of the current paylod:

```
[
  {
    "id": "aeafd489-ee98-4b2e-b77f-2aac2594b4b2",
    "topic": null,
    "subject": "durable/orchestrator/Running",
    "data": {
      "hubName": "DFHub",
      "functionName": "Counter",
      "instanceId": "13b602a9d2914d7c981016acbee0b941",
      "reason": "",
      "eventType": 0
    },
    "eventType": "orchestratorEvent",
    "eventTime": "2018-04-29T04:41:43.5299974Z",
    "metadataVersion": null,
    "dataVersion": "1.0"
  }
]
```

This pull request change it like this. 

```
[
  {
    "id": "aeafd489-ee98-4b2e-b77f-2aac2594b4b2",
    "topic": null,
    "subject": "durable/orchestrator/Running",
    "data": {
      "hubName": "DFHub",
      "functionName": "Counter",
      "instanceId": "13b602a9d2914d7c981016acbee0b941",
      "reason": "",
      "runtimeStatus": "Running"
    },
    "eventType": "orchestratorEvent",
    "eventTime": "2018-04-29T04:41:43.5299974Z",
    "metadataVersion": null,
    "dataVersion": "1.0"
  }
]
```
@cgillum , However, I have local unit testing problem. Please Full CI for this pull request. 
I tested in my environment, the all tests related this change is all pass. However, some test fails for Timeout.  (Obviously, it is not related this change). 

Failed Tests are

* DurableTaskEndToEndTests
* HandledActivityException
* OrchestrationConcurrency
* TimerCancellation

Please have a look.

### CODE REVIEW

File: src/WebJobs.Extensions.DurableTask/LifeCycleNotificationHelper.cs [hunk 1]
```diff
@@ -210,7 +210,7 @@ public void SetHttpMessageHandler(HttpMessageHandler handler)
                         FunctionName = functionName,
                         InstanceId = instanceId,
                         Reason = reason,
-                        EventType = orchestrationRuntimeStatus,
+                        RuntimeStatus = $"{orchestrationRuntimeStatus}",
```
Comment (cgillum) [inline src/WebJobs.Extensions.DurableTask/LifeCycleNotificationHelper.cs]: `orchestrationRuntimeStatus.ToString()` is better from a performance perspective.

Comment (TsuyoshiUshio) [reply src/WebJobs.Extensions.DurableTask/LifeCycleNotificationHelper.cs]: Thanks, I've done.

**Comment by cgillum** [comment]
@TsuyoshiUshio it seems your PRs are very unlucky - I don't normally see as many flaky test failures. :) I'm re-running the CI to see if everything passes this time.

**Comment by TsuyoshiUshio** [comment]
Sure. Or  you can change the Timeout of these tests. It  is just 10 sec.  I locally test it like this. as you said the performance should much better. 

```
            var s = new Stopwatch();
            var status01 = OrchestrationRuntimeStatus.Running;
            var status02 = OrchestrationRuntimeStatus.Completed;
            s.Start();
            for (int i = 0; i < 1000000; i++)
            {
                string c = $"{status01}";
                string d = $"{status02}";
            }
            Console.WriteLine("{xxx} : " + s.ElapsedMilliseconds);
            s.Stop();
            s.Restart();
            for (int i = 0; i < 1000000; i++)
            {
                string b = status01.ToString();
                string d = status02.ToString();
            }

            Console.WriteLine("ToString(): " + s.ElapsedMilliseconds);
            Console.ReadLine();
```
Result

```
{xxx} : 1190
ToString(): 385
```

**Comment by TsuyoshiUshio** [comment]
Other idea for the build time and flakiness, Bing team did a very good job for that. We might talk with them. :) Their build took a couple of hours in local machine. Now they have a bunch of tests, however, they can build test everyhing within 10 - 15 min. They said that their solution becomes available. 
https://stories.visualstudio.com/bing-continuous-delivery/

**Comment by TsuyoshiUshio** [comment]
Also, I'm gonna read the source related EndToEndTest related one.

**Comment by cgillum** [comment]
One thing I noticed when trying to improve robustness a few days ago is that there can be a lot of messages leftover in the queues from previous CI runs. Because task hub names are reused, the test code will often try to execute these old messages, causing unplanned overhead. Deleting these messages often makes the tests more reliable. That's what I did before the last CI restart and now it looks green.

So I think one thing we need to do to improve this situation is find a way to reliably clean up resources after test runs. However, profiling may still be useful because even if there are old messages, I would expect the tests to complete successfully.

**Pull request merged**